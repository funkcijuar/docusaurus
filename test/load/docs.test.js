import loadDocs from '@lib/load/docs.js';
import path from 'path';

describe('loadDocs', () => {
  test('simple docs', async () => {
    const docsDir = path.join(__dirname, '__fixtures__', 'simple-docs');
    const docsData = await loadDocs(docsDir);
    expect(docsData).toMatchSnapshot();
    expect(docsData).not.toBeNull();
  });

  test('no docs', async () => {
    const nonExistingDocsDir = path.join(
      __dirname,
      '__fixtures__',
      'nonExistingDocs'
    );
    const docsData = await loadDocs(nonExistingDocsDir);
    expect(docsData).toEqual([]);
  });
});
