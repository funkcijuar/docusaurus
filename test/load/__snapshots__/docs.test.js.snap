// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loadDocs simple docs 1`] = `
Array [
  Object {
    "id": "hello",
    "path": "/hello",
    "source": "hello.md",
    "title": "Hello, World !",
  },
  Object {
    "id": "bar",
    "path": "/foo/bar",
    "source": "foo/bar.md",
    "title": "Bar",
  },
  Object {
    "id": "baz",
    "path": "/foo/baz",
    "source": "foo/baz.md",
    "title": "baz",
  },
]
`;
